### PR TITLE
[server] remove multi-key partitions from cache correctly

### PR DESCRIPTION
**Pull request by zuston**
### Purpose

Fix incorrect Auto Partition cache removal for tables with multiple partition keys.

### Brief change log

- Fixed partition removal from the nested cache structure.
- Removed empty time-based cache entries.
- Added a regression test for multi-key partition removal.
- Resolved the test file merge conflict while preserving existing coverage.

### CODE REVIEW

File: fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java [hunk 1]
```diff
@@ -673,6 +687,30 @@ protected Integer getAutoCreateDayDelayMinutes(long tableId) {
...
+                    if (partitionMap == null) {
+                        return null;
+                    }
+
+                    Map<String, Set<String>> snapshot = new TreeMap<>();
+                    partitionMap.forEach(
+                            (partitionTime, partitionSet) ->
+                                    snapshot.put(
+                                            partitionTime,
+                                            partitionSet == null
```
Comment (Jackeyzhe) [inline fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java:714]: Nice fix, nit: could you add a test covering `partitionSet == null` branch?

**Comment by zuston** [comment]
cc @swuferhong

File: fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java (hunk 1 above)
Comment (zuston) [reply fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java:714]: make sense

File: fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java [hunk 2]
```diff
@@ -354,7 +354,21 @@ public void removePartition(long tableId, String partitionName) {
                 lock,
                 () -> {
                     if (autoPartitionTables.containsKey(tableId)) {
```
Comment (fresh-borzoni) [inline fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java:356]: addPartition just above calls addPartitionToPartitionsByTable. This does the inverse, but inline, so the "one partition key vs many" check now lives in three places: here, extractAutoPartitionValue, and addPartitionToPartitionsByTable. 
Can we add a removePartitionFromPartitionsByTable helper next to the add one so the two sides stay in sync?

File: fluss-server/src/test/java/org/apache/fluss/server/coordinator/AutoPartitionManagerTest.java [hunk 3]
```diff
@@ -677,6 +677,41 @@ void testDayPartitionDropShouldNotBeDelayedByJitter() throws Exception {
         assertThat(partitions.keySet()).contains("20250420", "20250421", "20250422");
     }
 
+    @Test
+    void testRemovePartitionFromMultiplePartitionKeysTable() throws Exception {
```
Comment (fresh-borzoni) [inline fluss-server/src/test/java/org/apache/fluss/server/coordinator/AutoPartitionManagerTest.java]: This table uses num-retention = 2, and with retention on the stale entry gets cleaned up by the next retention pass regardless. The case where it's never cleaned is num-retention = -1, where dropPartitions returns before touching the map. Could we cover that one too?

**Review by fresh-borzoni — COMMENTED** [review]
@zuston Thank you for the PR, left a couple of comments, PTAL

File: fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java (hunk 2 above)
Comment (zuston) [reply fluss-server/src/main/java/org/apache/fluss/server/coordinator/AutoPartitionManager.java:356]: nice suggestion!

File: fluss-server/src/test/java/org/apache/fluss/server/coordinator/AutoPartitionManagerTest.java (hunk 3 above)
Comment (fresh-borzoni) [reply fluss-server/src/test/java/org/apache/fluss/server/coordinator/AutoPartitionManagerTest.java]: @zuston Sorry, take back my -1 ask. It's not something a test can check and as written it does nothing anyway, since the test never triggers the scheduler. 
Let's put it back to @Test, please

**Review by fresh-borzoni — COMMENTED** [review]
@zuston Thank you, LGTM 👍 

cc @swuferhong

File: fluss-server/src/test/java/org/apache/fluss/server/coordinator/AutoPartitionManagerTest.java (hunk 3 above)
Comment (zuston) [reply fluss-server/src/test/java/org/apache/fluss/server/coordinator/AutoPartitionManagerTest.java]: +1. removed

**Review by fresh-borzoni — APPROVED** [review]
@zuston Thank you, LGTM 👍

**Comment by fresh-borzoni** [comment]
@zuston Thanks for the reminder, merging
